FTR, the error doesn't tell much:

```
Collecting rpm-py-installer (from -r requirements-dev.txt (line 10))
  Using cached rpm-py-installer-1.2.0.tar.gz (21 kB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'error'
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [76 lines of output]
      running egg_info
      writing rpm_py_installer.egg-info/PKG-INFO
      writing dependency_links to rpm_py_installer.egg-info/dependency_links.txt
      writing top-level names to rpm_py_installer.egg-info/top_level.txt
      reading manifest file 'rpm_py_installer.egg-info/SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      warning: no previously-included files found matching 'test-requirements.txt'
      no previously-included directories found matching 'tests'
      adding license file 'LICENSE'
      writing manifest file 'rpm_py_installer.egg-info/SOURCES.txt'
      Package: libpopt0
      Status: install ok installed
      Priority: important
      Section: libs
      Installed-Size: 120
      Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
      Architecture: amd64
      Multi-Arch: same
      Source: popt
      Version: 1.16-14
      Depends: libc6 (>= 2.27)
      Description: lib for parsing cmdline parameters
       Popt was heavily influenced by the getopt() and getopt_long() functions,
       but it allows more powerful argument expansion. It can parse arbitrary
       argv[] style arrays and automatically set variables based on command
       line arguments. It also allows command line arguments to be aliased via
       configuration files and includes utility functions for parsing arbitrary
       strings into argv[] arrays using shell-like rules.
       .
       This package contains the runtime library and locale data.
      Original-Maintainer: Michael Jeanson <mjeanson@debian.org>
      Homepage: http://rpm5.org/
      Get:1 http://azure.archive.ubuntu.com/ubuntu focal/main amd64 libpopt-dev amd64 1.16-14 [46.6 kB]
      Fetched 46.6 kB in 0s (611 kB/s)
      [INFO] Installing...
      [INFO] Created working directory '/tmp/tmptna7cqet-rpm-py-installer'
      [INFO] Downloading archive. 'https://github.com/rpm-software-management/rpm/archive/rpm-4.14.2.1-release.tar.gz'.
      Traceback (most recent call last):
        File "/tmp/pip-install-f__3ke5r/rpm-py-installer_86ad3148e43a4a6c85ee391e333f7730/install.py", line 2224, in <module>
          main()
        File "/tmp/pip-install-f__3ke5r/rpm-py-installer_86ad3148e43a4a6c85ee391e333f7730/install.py", line 2219, in main
          app.run()
        File "/tmp/pip-install-f__3ke5r/rpm-py-installer_86ad3148e43a4a6c85ee391e333f7730/install.py", line 46, in run
          raise InstallError(message)
      __main__.InstallError: RPM Python binding failed to install with unknown reason.
      Traceback (most recent call last):
        File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
        File "/tmp/pip-build-env-bi4knqk9/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 338, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
        File "/tmp/pip-build-env-bi4knqk9/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 320, in _get_build_requires
          self.run_setup()
        File "/tmp/pip-build-env-bi4knqk9/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 484, in run_setup
          super(_BuildMetaLegacyBackend,
        File "/tmp/pip-build-env-bi4knqk9/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 335, in run_setup
          exec(code, locals())
        File "<string>", line 94, in <module>
        File "/tmp/pip-build-env-bi4knqk9/overlay/lib/python3.9/site-packages/setuptools/__init__.py", line 108, in setup
          return distutils.core.setup(**attrs)
        File "/tmp/pip-build-env-bi4knqk9/overlay/lib/python3.9/site-packages/setuptools/_distutils/core.py", line 185, in setup
          return run_commands(dist)
        File "/tmp/pip-build-env-bi4knqk9/overlay/lib/python3.9/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
          dist.run_commands()
        File "/tmp/pip-build-env-bi4knqk9/overlay/lib/python3.9/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands
          self.run_command(cmd)
        File "/tmp/pip-build-env-bi4knqk9/overlay/lib/python3.9/site-packages/setuptools/dist.py", line 1221, in run_command
          super().run_command(command)
        File "/tmp/pip-build-env-bi4knqk9/overlay/lib/python3.9/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "<string>", line 57, in run
        File "<string>", line 20, in install_rpm_py
      Exception: Command failed: /opt/hostedtoolcache/Python/3.9.16/x64/bin/python install.py
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
make: *** [Makefile:91: test-setup] Error 1
```

But the reality is that `rpm-py-installer` intentionally blocks `wheel` installs in https://github.com/junaruga/rpm-py-installer/blob/ba1520688d8f7cd46dad6fe6b4192ce8516f0ac3/setup.py#L83-L89

See also https://github.com/junaruga/rpm-py-installer/issues/276

--- a/Makefile
+++ b/Makefile
@@ -87,7 +87,7 @@ clean_%: FORCE $(MANIFEST)
 setup: test-setup
 
 test-setup: | tests/test_playbooks/vars/server.yml
-	pip install --upgrade pip
+	pip install --upgrade --force-reinstall 'pip<23.1'
 	pip install --upgrade -r requirements-dev.txt
 
 tests/test_playbooks/vars/server.yml:

